R/ncbi_snp_api.R

```diff
@@ -3,52 +3,61 @@
+                                          one_allele$allele$spdi$inserted_sequence,
+                                          sep = ",")
+                        if (one_allele$allele$spdi$inserted_sequence == "") {
+                          VariationAllele <-  paste0("del",
+                                                     one_allele$allele$spdi$deleted_sequence)
+                          alleles <-  paste(alleles, VariationAllele)
+                        } else {
+                          VariationAllele <- one_allele$allele$spdi$inserted_sequence
                         }
-      )
-      
-      ## remove the NULLs
-      df_list <- df_list[lengths(df_list) != 0]
-      
-      ## if more than one we need to merge the ancestral and variation alleles
-      df_all <- do.call(rbind, df_list)
-      new_df <- stats::aggregate(. ~ BP, data = df_all,
-                                 FUN = function(x)
-                                   paste(unique(x[x != ""]),
-                                         collapse = ",")
-      )
-      new_df$Alleles <- paste(unique(unlist(strsplit(new_df$Alleles, ","))),
-                              collapse = ",")
-      return(new_df)
+                        
+                        df_snp <- data.frame(Alleles = alleles,
+                                             AncestralAllele = one_allele$allele$spdi$deleted_sequence,
+                                             VariationAllele = VariationAllele,
+                                             BP = one_allele$allele$spdi$position + 1,
+                                             seqname = one_allele$allele$spdi$seq_id,
+                                             hgvs = one_allele$hgvs,
+                                             assembly = assembly_name,
+                                             Chromosome = chrom_name,
+                                             stringsAsFactors = FALSE)
+                        return(df_snp)
+                      }
+                    }
+  )
+  
+  ## remove the NULLs
+  df_list <- df_list[lengths(df_list) != 0]
+  
+  ## if more than one we need to merge the ancestral and variation alleles
+  df_all <- do.call(rbind, df_list)
+  new_df <- stats::aggregate(. ~ BP, data = df_all,
+                             FUN = function(x)
+                               paste(unique(x[x != ""]),
+                                     collapse = ",")
+  )
+  new_df$Alleles <- paste(unique(unlist(strsplit(new_df$Alleles, ","))),
+                          collapse = ",")
+  return(new_df)
+}
+
+get_placements <- function(primary_info, version) {
+  for (record in primary_info$primary_snapshot_data$placements_with_allele) {
+    if (record$is_ptlp & length(record$placement_annot$seq_id_traits_by_assembly) > 0 && version == "38") {
+      return(get_placements_inside(record))
+    }else if(length(record$placement_annot$seq_id_traits_by_assembly) > 0 && version == "37"){
```

Comment:
I am not so keen on the hard-coding of the versions here. I will have to play around with the code a bit to propose an improvement.